# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/04/01 05:02:39-05:00 len.brown@intel.com 
#   [ACPI] Restore PIC-mode SCI default to Level Trigger (David Shaohua Li)
#   http://bugme.osdl.org/show_bug.cgi?id=2382
# 
# drivers/acpi/bus.c
#   2004/04/01 04:59:33-05:00 len.brown@intel.com +5 -0
#   fix a regression where we stopped forcing PIC-mod
#   SCI to Level from Edge if "acpi_sci=level" was not used
# 
diff -Nru a/drivers/acpi/bus.c b/drivers/acpi/bus.c
--- a/drivers/acpi/bus.c	Thu Apr  1 05:02:48 2004
+++ b/drivers/acpi/bus.c	Thu Apr  1 05:02:48 2004
@@ -615,6 +615,11 @@
 #ifdef CONFIG_X86
 	if (!acpi_ioapic) {
 		extern acpi_interrupt_flags acpi_sci_flags;
+
+		/* compatible (0) means level (3) */
+		if (acpi_sci_flags.trigger == 0)
+			acpi_sci_flags.trigger = 3;
+
 		/* Set PIC-mode SCI trigger type */
 		acpi_pic_sci_set_trigger(acpi_fadt.sci_int, acpi_sci_flags.trigger);
 	} else {
